### PR TITLE
Sameera/fix matchings page

### PR DESCRIPTION
**Pull request by sameerav**

### CODE REVIEW

File: app/assets/javascripts/components/admin/matchings_components/student_description.es6.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  static get PropTypes() {
+    return {
+      student: React.PropTypes.object,
+      onClick: React.PropTypes.func
+    };
+  }
+
+  renderInstrumentButton(instrument) {
+    return (
+      <Button className="button button--outline-orange" onClick={(event) => this.props.onClick(this.props.student.id, instrument)}>{ instrument.name }</Button>
```
Comment (sonia-y) [inline app/assets/javascripts/components/admin/matchings_components/student_description.es6.jsx]: remove extra spaces inside brackets {instrument.name}

File: app/assets/javascripts/components/admin/matchings_components/student_description.es6.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    );
```
Comment (sonia-y) [inline app/assets/javascripts/components/admin/matchings_components/student_description.es6.jsx]: remove extra spaces inside brackets `{instrument.name}`

File: app/assets/javascripts/components/admin/matchings_components/student_description.es6.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  renderInstrumentButtons() {
+    return this.props.student.instruments.map((instrument) => this.renderInstrumentButton(instrument));
+  }
+
+  render () {
+    return (
+      <div className="person-description">
+        <p>Name: { this.props.student.first_name } { this.props.student.last_name }</p>
```
Comment (sonia-y) [inline app/assets/javascripts/components/admin/matchings_components/student_description.es6.jsx]: same, remove spaces

File: app/controllers/api/teachers_controller.rb [hunk 4]
```diff
@@ -56,20 +56,26 @@ def recent_lessons
...
-      if (is_valid_matching(teacher, student) && teacher.is_searching)
+      if (is_valid_matching(teacher, student, instrument) && teacher.is_searching)
         teachers.push(teacher)
       end
     end
     render json: teachers
   end
 
-  def is_valid_matching(teacher, student)
+  def is_valid_matching(teacher, student, instrument)
```
Comment (sonia-y) [inline app/controllers/api/teachers_controller.rb]: wow so cool :)

File: app/controllers/api/students_controller.rb [hunk 5]
```diff
@@ -55,20 +55,26 @@ def recent_lessons
...
-    for matching in matchings do
-      student_ids.add matching.student_id
-    end
-
     all_students = Student.all
     students = []
-    all_students.each do |student|
-      if (!(student_ids.include? student.id))
+    for student in all_students
+      fully_matched = true
```
Comment (sonia-y) [inline app/controllers/api/students_controller.rb]: change these for loops

File: app/controllers/api/students_controller.rb [hunk 6]
```diff
@@ -55,20 +55,26 @@ def recent_lessons
...
+        if not matched_for_instrument
+          fully_matched = false
+        end
+      end
+      if fully_matched
         students.push(student)
       end
     end
-    render json: students
+    render json: students.to_json(include: :instruments)
```
Comment (sonia-y) [inline app/controllers/api/students_controller.rb]: hm, wait if you do render json: you can just use the serializer!!

File: app/assets/javascripts/components/admin/unmatched_page.es6.jsx [hunk 7]
```diff
@@ -27,11 +28,12 @@ class UnmatchedPage extends React.Component {
     );
   }
 
-  studentOnClick(studentId) {
+  studentOnClick(studentId, instrument) {
     var studentRoute = ApiConstants.students.show(studentId)
     var studentResolve = ((response) => {
       this.setState({fullStudent: true, student: response});
-      var teacherRoute = ApiConstants.teachers.possibleTeachers(studentId);
+      this.setState({instrument: instrument.name});
```
Comment (sonia-y) [inline app/assets/javascripts/components/admin/unmatched_page.es6.jsx]: combine this setstate with the one on top so 

`this.setState({ 
       fullStudent: true, 
        student: response,
        instrument: instrument.name,
})`

Comment (sonia-y) [inline app/assets/javascripts/components/admin/unmatched_page.es6.jsx]: except, put it in different lines

File: app/controllers/api/teachers_controller.rb (hunk 4 above)
Comment (sameerav) [inline app/controllers/api/teachers_controller.rb]: same

File: app/assets/javascripts/components/admin/matchings_components/teacher_description.es6.jsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    return {
+      teacher: React.PropTypes.object,
+      onClick: React.PropTypes.func
+    };
+  }
+
+  render () {
+    return (
+      <div className="person-description">
+        <p>Name: { this.props.teacher.first_name } { this.props.teacher.last_name }</p>
```
Comment (sonia-y) [inline app/assets/javascripts/components/admin/matchings_components/teacher_description.es6.jsx]: remove spacing here too!!

File: app/assets/javascripts/components/admin/matchings_components/teacher_description.es6.jsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+        <p>City: { this.props.teacher.city }</p>
```
Comment (sonia-y) [inline app/assets/javascripts/components/admin/matchings_components/teacher_description.es6.jsx]: here too

File: app/assets/javascripts/components/admin/unmatched_page.es6.jsx [hunk 10]
```diff
@@ -27,11 +28,13 @@ class UnmatchedPage extends React.Component {
     );
   }
 
-  studentOnClick(studentId) {
+  studentOnClick(studentId, instrument) {
     var studentRoute = ApiConstants.students.show(studentId)
     var studentResolve = ((response) => {
-      this.setState({fullStudent: true, student: response});
-      var teacherRoute = ApiConstants.teachers.possibleTeachers(studentId);
+      this.setState({fullStudent: true,
```
Comment (sonia-y) [inline app/assets/javascripts/components/admin/unmatched_page.es6.jsx]: put first attribute in new line 
so 
this.setState({
    fullStudent: true,
   });

**Comment by sonia-y** [comment]
sorry 2 mini syntax things and then ur good

**Pull request merged**